pmd-core/src/main/java/net/sourceforge/pmd/properties/EnumeratedProperty.java

```diff
@@ -159,4 +159,53 @@ private EnumPBuilder(String name) {
         }
     }
 
+    public static class EnumeratedPropertyBuilder<E> {
```

Comment:
Lol did you miss the existing builders? You can get one using eg `EnumeratedProperty.<E>named("name")`. I hope that was generated code lol :)

Reply:
Haha no, it wasn't generated code, but was easy to write. In the case of EnumPBuilder, to create a new EnumeratedProperty, it uses the fields from the EnumeratedProperty instance, so it wasn't the use case that i needed. In StringPBuilder, the constructor is private and uses instance fields from the StringProperty instance.

That's why I decided to write an "instance-independent" implementation.

Reply:
The fields (this.name, this.description, etc.) are not the fields of an EnumeratedProperty (the class is static). They're instead inherited from the hierarchy of [PropertyDescriptorBuilder](https://github.com/pmd/pmd/blob/0fb9c4678c92837981762f93e44ed0bab7097444/pmd-core/src/main/java/net/sourceforge/pmd/properties/builders/PropertyDescriptorBuilder.java#L24). 

The constructor is private on purpose, like in other property+builder implementations, so that the only way to retrieve an instance of a builder is by calling static factory methods like `EnumeratedProperty.named(String)`. This is a use case of the builder pattern to ensure that a least the name argument is provided. 

Please use the method `named` and remove these duplicate implementations

Reply:
You're right, I totally skipped that...

Reply:
Fixed. I'd never seen something like that, it seems like an interesting design case of study, thanks for the insight!

Reply:
You can learn more about the builder pattern in [Effective Java Item 2](https://www.informit.com/articles/article.aspx?p=1216151&seqNum=2) ;)